aspnetcore/security/cors.md

```diff
@@ -42,110 +42,97 @@ These URLs have different origins than the previous two URLs:
 
-Use `[EnableCors]` to specify the default policy and `[EnableCors("{Policy String}")]` to specify a policy.
+The [[EnableCors]](xref:Microsoft.AspNetCore.Cors.EnableCorsAttribute) attribute provides an alternative to applying CORS globally. The `[EnableCors]` attribute enables CORS for selected endpoints, rather than all endpoints:
+
+* `[EnableCors]` specifies the default policy.
+* `[EnableCors("{Policy String}")]` specifies a specific a policy.
 
 The `[EnableCors]` attribute can be applied to:
 
 * Razor Page `PageModel`
 * Controller
 * Controller action method
 
-You can apply different policies to controller/page-model/action with the  `[EnableCors]` attribute. When the `[EnableCors]` attribute is applied to a controllers/page-model/action method, and CORS is enabled in middleware, both policies are applied. We recommend against combining policies. Use the `[EnableCors]` attribute or middleware, not both in the same app.
+Different policies can be applied to controllers, page models, or action methods with the `[EnableCors]` attribute. When the `[EnableCors]` attribute is applied to a controller, page model, or action method, and CORS is enabled in middleware, ***both*** policies are applied. ***We recommend against combining policies. Use the*** `[EnableCors]` ***attribute or middleware, not both in the same app.***
 
 The following code applies a different policy to each method:
 
-[!code-csharp[](cors/sample/Cors/WebAPI/Controllers/WidgetController.cs?name=snippet&highlight=6,14)]
+[!code-csharp[](cors/3.1sample/Cors/WebAPI/Controllers/WidgetController.cs?name=snippet&highlight=6,14)]
 
 The following code creates a CORS default policy and a policy named `"AnotherPolicy"`:
 
-[!code-csharp[](cors/sample/Cors/WebAPI/StartupMultiPolicy.cs?name=snippet&highlight=12-28)]
+[!code-csharp[](cors/3.1sample/Cors/WebAPI/Startup3.cs?name=snippet&highlight=12-28,44)]
+
+See [Test CORS](#testc) for instructions on testing code similar to the preceding code.
 
 ### Disable CORS
 
-The [&lbrack;DisableCors&rbrack;](xref:Microsoft.AspNetCore.Cors.DisableCorsAttribute) attribute disables CORS for the controller/page-model/action.
+The [[DisableCors]](xref:Microsoft.AspNetCore.Cors.DisableCorsAttribute) attribute does ***not***  disables CORS when a CORS policy has been enabled in middleware. For example, either of the following calls in `Startup.Configure` sets a CORS policy in middleware:
+
+* `app.UseCors("MyPolicy");`
+* `endpoints.MapControllers().RequireCors("MyPolicy");`
+
+The following code doesn't set a CORS policy in middleware:
+
+[!code-csharp[](cors/3.1sample/Cors/WebAPI/Startup3.cs?name=snippet2)]
+
+Using the preceding code, the following code disables CORS in for the `GetValues2` method:
+
+[!code-csharp[](cors/3.1sample/Cors/WebAPI/Controllers/ValuesController.cs?name=snippet&highlight=1,23)]
+
+See [Test CORS](#testc) for instructions on testing the preceding code.
```

Comment:
I think this section should be ordered something like this:

1. Explain what `[DisableCors]` does.
2. Show the second snippet, explaining how `[DisableCors]` on the action overrides `[EnableCors]` on the controller.
3. Discuss how `[DisableCors]` works with `UseCors`. As I've called out above, this behaviour differs between endpoint routing and "not" endpoint routing.

I don't think there's any value in the first snippet, especially as it's still possible to set a default policy using the `AddCors` call.

Reply:
@serpent5 I think I addressed this. Can you review the public link?